server/routes.go

```diff
@@ -1492,7 +1495,36 @@ func (s *Server) ChatHandler(c *gin.Context) {
 				res.LoadDuration = checkpointLoaded.Sub(checkpointStart)
 			}
 
-			ch <- res
+			// TODO: Consolidate streaming and non-streaming request patterns
+			if req.Stream != nil && !*req.Stream {
+				res.Message.Content = r.Content
+				ch <- res
+				return
+			}
+
+			if len(req.Tools) == 0 {
+				ch <- res
+				return
+			}
+
+			// If tools are recognized, use a flag to track the sending of a tool downstream
+			// This ensures that content is cleared from the message on the last chunk sent
```

Comment:
"attempt to parse tool calls, sending a chunk whenever a complete tool call is parsed"

Reply:
(or something along those lines)